### PR TITLE
Expose opentracing start span options to allow tag customization

### PR DESCRIPTION
**Pull request by gruane-sq**
Exposes the `opentracing.StartSpanOption` to allow callers to pass custom span tags. One use case is to allow control over [Datadog's Unified Span Tagging](https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/?tab=kubernetes). This is commonly done with database clients so that queries show as a different "service" in the Datadog interface, with their own [Service Page](https://docs.datadoghq.com/tracing/services/service_page/) that is tailored specifically to database interactions.

### CODE REVIEW

**Review by ryho — APPROVED** [review]

File: db.go [hunk 1]
```diff
@@ -390,10 +390,12 @@ type DB struct {
 	//
 	// The default is false, tracers should be registered by the callers.
 	OpentracingEnabled bool
-	context            context.Context
-	mu                 *sync.RWMutex // pointer, so copies in WithContext() behave properly.
-	models             map[reflect.Type]*Model
-	mappings           map[reflect.Type]fieldMap
+	// Span option functions are applied to every span started by squalor, when OpentracingEnabled is true.
+	SpanOptions []opentracing.StartSpanOption
```
Comment (danrice67) [inline db.go:394]: You made this a slice but only allow a single option to be set

File: db.go [hunk 2]
```diff
@@ -502,6 +504,17 @@ func SetOpentracingEnabled(enabled bool) DBOption {
 	}
 }
 
+// SetStartSpanOption will apply the span option function to every opentracing span created by squalor
+func SetStartSpanOption(spanOption opentracing.StartSpanOption) DBOption {
+	return func(db *DB) error {
+		if db.SpanOptions == nil {
```
Comment (danrice67) [inline db.go:510]: You don't need the `if` since `append` will create the slice if needed

Comment (danrice67) [reply db.go:510]: You could also make this function take varargs `...opentracing.StartSpanOption` and append them all

File: db.go (hunk 1 above)
Comment (danrice67) [inline db.go:394]: Does this need to be visible (capitalized)?

File: db.go [hunk 3]
```diff
@@ -513,6 +526,7 @@ func NewDB(db *sql.DB, options ...DBOption) (*DB, error) {
 		IgnoreMissingCols:     false,
 		Logger:                nil,
 		OpentracingEnabled:    false,
+		SpanOptions:           make([]opentracing.StartSpanOption, 0),
```
Comment (danrice67) [inline db.go:529]: Could probably initialize to `nil` just fine

File: db.go (hunk 2 above)
Comment (gruane-sq) [reply db.go:510]: Good idea - fixed

File: db.go (hunk 1 above)
Comment (gruane-sq) [reply db.go]: nope - made private

File: db.go (hunk 3 above)
Comment (gruane-sq) [reply db.go]: updated

**Pull request merged**